src/t8_forest/t8_forest_iterate.cxx

```diff
@@ -426,15 +426,127 @@ t8_forest_iterate_replace (t8_forest_t forest_new,
     T8_ASSERT (eclass == t8_forest_get_tree_class (forest_old, itree));
     ts = t8_forest_get_eclass_scheme (forest_new, eclass);
     T8_ASSERT (ts == t8_forest_get_eclass_scheme (forest_new, eclass));
-    for (ielem_new = 0, ielem_old = 0; ielem_new < elems_per_tree_new
-         || ielem_old < elems_per_tree_old;) {
+
+
+    if (elems_per_tree_new == 0) {
+        /* all elements of (new) tree got removed */
+        for (ielem_old = 0; ielem_old < elems_per_tree_old; ielem_old++) {
+          replace_fn (forest_old, forest_new, itree, ts, 1, ielem_old,
+                      0, NULL);
+        }      
+    }
+    else {
+      T8_ASSERT(elems_per_tree_old > 0 && elems_per_tree_new > 0);
+
+      for (ielem_new = 0, ielem_old = 0; ielem_new < elems_per_tree_new
+          || ielem_old < elems_per_tree_old; ) {
+      
+      T8_ASSERT(ielem_new <= elems_per_tree_new);
+      T8_ASSERT(ielem_old <= elems_per_tree_old);
+
       /* Iterate over the elements */
       /* Get pointers to the elements */
       elem_new = t8_forest_get_element_in_tree (forest_new, itree, ielem_new);
       elem_old = t8_forest_get_element_in_tree (forest_old, itree, ielem_old);
+      
       /* Get the levels of these elements */
       level_new = ts->t8_element_level (elem_new);
       level_old = ts->t8_element_level (elem_old);
+
+      /* Get nearest common ancestor */
+      ts->t8_element_nca(elem_new, elem_old, elem_nca);
```

Comment:
I don't think, that it is necessary to use the nca here. If we can interpolate between ```elem_new``` and ```elem_old``` they should be in parent/child-relation.